src/Stratis.Bitcoin/FullNode.cs

```diff
@@ -215,13 +215,14 @@ public void Stop()
             // Fire INodeLifetime.Stopping.
             this.nodeLifetime.StopApplication();
 
-            this.ConnectionManager.Dispose();
+            // Fire the NodeFeatureExecutor.Stop.
+            this.fullNodeFeatureExecutor.Stop();
 
             foreach (IDisposable disposable in this.Resources)
                 disposable.Dispose();
 
-            // Fire the NodeFeatureExecutor.Stop.
-            this.fullNodeFeatureExecutor.Stop();
+            this.ConnectionManager.Dispose();
+            this.NodeService<NodeSettings>().LoggerFactory.Dispose();
```

Comment:
You could inject settings in the constructor similar to ConnetcionManager

Reply:
It doesn't have a constructor and we are getting CM this way:
`this.ConnectionManager = this.Services.ServiceProvider.GetService<IConnectionManager>();`

Would you like me to add constructor to full node and use injection for getting the settings?

Reply:
Yes, just create a private field:
```
        /// <summary>Logger factory to create loggers.</summary>
        private readonly ILoggerFactory loggerFactory;
```

And set it in `Initialize`, then you can dispose of it properly in `Stop `.

Reply:
Fixed.